src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1069 @@
+# encoding: utf-8
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+#
+# Module: Manipulate system authentication configuration.
+# Author: Howard Guo <hguo@suse.com>
+
+require 'yast'
+require 'resolv'
+require 'open3'
+require 'socket'
+require 'tempfile'
+require 'fileutils'
+require 'date'
+require 'auth/krbparse'
+
+module Auth
+    # Manage system-wide authentication configuration from Kerberos, LDAP, Samba, and SSSD's perspectives.
```

Comment:
This class is too big.
Ideally a class should have only single responsibility. In practice that may be easier or harder to do. In this case it seems easy: split out classes for 
- Kerberos
- LDAP
- Samba
- SSSD

and have Auth::AuthConf mediate between them (if needed at all).